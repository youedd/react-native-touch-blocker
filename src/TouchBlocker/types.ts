import { ViewProps } from 'react-native'

export type TouchBlockerProps = ViewProps & {
  enableTouchEvents?: boolean
}
