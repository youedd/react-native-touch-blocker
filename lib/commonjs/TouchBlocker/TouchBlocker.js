"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _reactNative = require("react-native");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

const panResponder = _reactNative.PanResponder.create({
  onPanResponderTerminationRequest: () => false,
  onStartShouldSetPanResponderCapture: () => true,
  onShouldBlockNativeResponder: () => true
});

const TouchBlocker = _ref => {
  let {
    enableTouchEvents = false,
    ...props
  } = _ref;
  const panHanlders = enableTouchEvents ? {} : panResponder.panHandlers;
  return /*#__PURE__*/_react.default.createElement(_reactNative.View, _extends({}, props, panHanlders));
};

var _default = TouchBlocker;
exports.default = _default;
//# sourceMappingURL=TouchBlocker.js.map