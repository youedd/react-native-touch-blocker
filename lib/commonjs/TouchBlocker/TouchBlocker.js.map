{"version":3,"sources":["TouchBlocker.tsx"],"names":["panResponder","PanResponder","create","onPanResponderTerminationRequest","onStartShouldSetPanResponderCapture","onShouldBlockNativeResponder","TouchBlocker","enableTouchEvents","props","panHanlders","panHandlers"],"mappings":";;;;;;;AAAA;;AACA;;;;;;AAGA,MAAMA,YAAY,GAAGC,0BAAaC,MAAb,CAAoB;AACvCC,EAAAA,gCAAgC,EAAE,MAAM,KADD;AAEvCC,EAAAA,mCAAmC,EAAE,MAAM,IAFJ;AAGvCC,EAAAA,4BAA4B,EAAE,MAAM;AAHG,CAApB,CAArB;;AAMA,MAAMC,YAAyC,GAAG,QAG5C;AAAA,MAH6C;AACjDC,IAAAA,iBAAiB,GAAG,KAD6B;AAEjD,OAAGC;AAF8C,GAG7C;AACJ,QAAMC,WAAW,GAAGF,iBAAiB,GAAG,EAAH,GAAQP,YAAY,CAACU,WAA1D;AACA,sBAAO,6BAAC,iBAAD,eAAUF,KAAV,EAAqBC,WAArB,EAAP;AACD,CAND;;eAQeH,Y","sourcesContent":["import React from 'react'\nimport { PanResponder, View } from 'react-native'\nimport { TouchBlockerProps } from './types'\n\nconst panResponder = PanResponder.create({\n  onPanResponderTerminationRequest: () => false,\n  onStartShouldSetPanResponderCapture: () => true,\n  onShouldBlockNativeResponder: () => true\n})\n\nconst TouchBlocker: React.FC<TouchBlockerProps> = ({\n  enableTouchEvents = false,\n  ...props\n}) => {\n  const panHanlders = enableTouchEvents ? {} : panResponder.panHandlers\n  return <View {...props} {...panHanlders} />\n}\n\nexport default TouchBlocker\n"]}