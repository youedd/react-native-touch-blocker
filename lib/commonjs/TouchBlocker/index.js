"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
var _exportNames = {};
Object.defineProperty(exports, "default", {
  enumerable: true,
  get: function () {
    return _TouchBlocker.default;
  }
});

var _types = require("./types");

Object.keys(_types).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _types[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _types[key];
    }
  });
});

var _TouchBlocker = _interopRequireDefault(require("./TouchBlocker"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
//# sourceMappingURL=index.js.map