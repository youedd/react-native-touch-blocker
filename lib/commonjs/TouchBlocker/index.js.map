{"version":3,"sources":["index.ts"],"names":[],"mappings":";;;;;;;;;;;;;AAAA;;AAAA;AAAA;AAAA;AAAA;AAAA;AAAA;AAAA;AAAA;AAAA;AAAA;AAAA;;AACA","sourcesContent":["export * from './types'\nexport { default } from './TouchBlocker'\n"]}