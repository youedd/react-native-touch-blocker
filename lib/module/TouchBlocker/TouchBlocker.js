function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

import React from 'react';
import { PanResponder, View } from 'react-native';
const panResponder = PanResponder.create({
  onPanResponderTerminationRequest: () => false,
  onStartShouldSetPanResponderCapture: () => true,
  onShouldBlockNativeResponder: () => true
});

const TouchBlocker = _ref => {
  let {
    enableTouchEvents = false,
    ...props
  } = _ref;
  const panHanlders = enableTouchEvents ? {} : panResponder.panHandlers;
  return /*#__PURE__*/React.createElement(View, _extends({}, props, panHanlders));
};

export default TouchBlocker;
//# sourceMappingURL=TouchBlocker.js.map