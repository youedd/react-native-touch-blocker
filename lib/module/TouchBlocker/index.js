export * from './types';
export { default } from './TouchBlocker';
//# sourceMappingURL=index.js.map