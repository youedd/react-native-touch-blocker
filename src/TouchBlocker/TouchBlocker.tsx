import React from 'react'
import { PanResponder, View } from 'react-native'
import { TouchBlockerProps } from './types'

const panResponder = PanResponder.create({
  onPanResponderTerminationRequest: () => false,
  onStartShouldSetPanResponderCapture: () => true,
  onShouldBlockNativeResponder: () => true
})

const TouchBlocker: React.FC<TouchBlockerProps> = ({
  enableTouchEvents = false,
  ...props
}) => {
  const panHanlders = enableTouchEvents ? {} : panResponder.panHandlers
  return <View {...props} {...panHanlders} />
}

export default TouchBlocker
