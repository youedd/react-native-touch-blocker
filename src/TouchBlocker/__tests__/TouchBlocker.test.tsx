import { render } from '@testing-library/react-native'
import React from 'react'
import { Button } from 'react-native'
import TouchBlocker from '../TouchBlocker'

describe('TouchBlocker Component unit tests :', () => {
  it('Should render children ', () => {
    const mockFn = jest.fn()
    const { toJSON } = render(
      <TouchBlocker>
        <Button title='button' onPress={mockFn} />
      </TouchBlocker>
    )

    const tree = toJSON()
    expect(tree).toMatchSnapshot()
  })

  it("Shouldn't add panHandlers to Root view when enableTouchEvents prop is true", () => {
    const mockFn = jest.fn()
    const { toJSON } = render(
      <TouchBlocker enableTouchEvents>
        <Button title='button' onPress={mockFn} />
      </TouchBlocker>
    )

    const tree = toJSON()
    expect(tree).toMatchSnapshot()
  })
})
