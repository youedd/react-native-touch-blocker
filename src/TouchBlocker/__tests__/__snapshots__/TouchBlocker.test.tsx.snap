// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TouchBlocker Component unit tests : Should render children  1`] = `
<View
  onMoveShouldSetResponder={[Function]}
  onMoveShouldSetResponderCapture={[Function]}
  onResponderEnd={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderReject={[Function]}
  onResponderRelease={[Function]}
  onResponderStart={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  onStartShouldSetResponderCapture={[Function]}
>
  <View
    accessibilityRole="button"
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        button
      </Text>
    </View>
  </View>
</View>
`;

exports[`TouchBlocker Component unit tests : Shouldn't add panHandlers to Root view when enableTouchEvents prop is true 1`] = `
<View>
  <View
    accessibilityRole="button"
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        button
      </Text>
    </View>
  </View>
</View>
`;
